pkg/envoy/xdsutil.go

```diff
@@ -396,3 +398,21 @@ func GetServiceAccountFromProxyCertificate(cn certificate.CommonName) (identity.
 
 	return svcAccount, nil
 }
+
+// IsMetricsEnabled returns true if the pod in the mesh is correctly annotated for prometheus scrapping
+func IsMetricsEnabled(proxy *Proxy, meshCatalog catalog.MeshCataloger) bool {
+	isScrapingEnabled := false
+	pod, err := GetPodFromCertificate(proxy.GetCertificateCommonName(), meshCatalog.GetKubecontroller())
```

Comment:
Could you instead implement this function in the `kubernetes.Controller` interface given this is very specific to k8s and now leaks k8s stuff even further into pkg/envoy?

Code in pkg/envoy can use:
`meshCatalog.kubeController.IsMetricsEnabled(pod)`

Reply:
Thanks for the suggestion, I moved the code into `pkg/kubernetes`